### PR TITLE
feat: persist optional `body` param for friend requests

### LINKED ISSUE #119: Persist optional body param for friend requests

### LINKED ISSUE #120: Get optional body param for friend requests

### PR DESCRIPTION
**Pull request by Julieta11**
Closes #119 
Closes #120

### CODE REVIEW

File: tests/routes/matrix/room_events.rs [hunk 1]
```diff
@@ -266,7 +266,7 @@ mod tests {
     }
 
     fn get_request(token: &str, event_type: FriendshipEvent) -> Request {
-        let body = RoomEventRequestBody { r#type: event_type };
+        let body = RoomEventRequestBody { r#type: event_type, body: None };
```
Comment (moliva) [inline tests/routes/matrix/room_events.rs:269]: Let's also add a test for a case where the body is present.

File: src/routes/synapse/room_events.rs [hunk 2]
```diff
@@ -123,11 +124,14 @@ pub async fn room_event_handler(
         (logged_in_user, token)
     };
 
+    let body_message = body.body.as_deref();
```
Comment (moliva) [inline src/routes/synapse/room_events.rs:127]: Wondering if we should change this to this version and work with an `Option<String>` instead. What do you think @jmoguilevsky @lauti7 @agusaldasoro ?
```suggestion
    let body_message = body.body.clone();
```

Comment (jmoguilevsky) [reply src/routes/synapse/room_events.rs:127]: I'm not sure I understood, it's already an Option<String>

Comment (moliva) [reply src/routes/synapse/room_events.rs:127]: `Option<String>`  vs `Option<&str>`

Comment (Julieta11) [reply src/routes/synapse/room_events.rs:127]: - [String vs &str; in Rust functions](https://web.archive.org/web/20201112023149/https://hermanradtke.com/2015/05/03/string-vs-str-in-rust-functions.html)
- [Creating a Rust function that accepts String or &str;](https://web.archive.org/web/20201112021433/https://hermanradtke.com/2015/05/06/creating-a-rust-function-that-accepts-string-or-str.html)

**Review by jmoguilevsky — APPROVED** [review]

**Pull request merged**